source/Calamari.Tests/Fixtures/Nginx/NginxFixture.cs

```diff
@@ -285,5 +294,33 @@ public void TestLocationsUnsuitableForFilenames()
             // Ensure the filename matches the expected format.
             Assert.IsTrue(Directory.GetFiles(Path.Combine(tempDirectory, "conf"))[0].EndsWith($"{nginxServer.VirtualServerName}.conf"));
         }
+
+        (string, string, string) GetCertificateDetails()
+        {
+#if NETCORE
+            var chainCertFilePath = TestEnvironment.GetTestPath("Helpers", "Certificates", "SampleCertificateFiles", "3-cert-chain.pfx");
+            var certificateCollection = new X509Certificate2Collection();
+            certificateCollection.Import(chainCertFilePath, "hello world", X509KeyStorageFlags.PersistKeySet);
+
+            var certificate = certificateCollection.First();
+            var certificatePem = new string(PemEncoding.Write("CERTIFICATE", certificate.RawData));
+            var key = (AsymmetricAlgorithm)certificate.GetRSAPrivateKey() ?? certificate.GetECDsaPrivateKey();
+            var privateKeyBytes = key.ExportPkcs8PrivateKey();
+            var certificatePrivateKeyPem = new string(PemEncoding.Write("PRIVATE KEY", privateKeyBytes));
+
+            string certificateChainPem = null;
+            foreach (var cert in certificateCollection.Skip(1))
+            {
+                // if the cert has no extension or the cert has a basic constraints extension,
+                // and the certificate authority is true then this is a chain certificate
+                certificateChainPem += $@"{new string(PemEncoding.Write("CERTIFICATE", cert.RawData))}
+";
+            }
+
+            return (certificatePem, certificatePrivateKeyPem, certificateChainPem);
+#else
+            return ("", "", "");
```

Comment:
nit: prefer `string.Empty` over `""`
There is no technical/perf reason for this, just readability and clarity of intent.